CHANGELOG.md

```diff
@@ -4,6 +4,210 @@ All notable changes to this project will be documented in this file.
+
+```rust
+let api = OnlineClient::<PolkadotConfig>::new().await?;
+
+let address = polkadot::constants()
+    .balances()
+    .existential_deposit();
+
+let existential_deposit = api.constants().at(&address)?;
+```
+
+See the `examples/examples/fetch_constants.rs` example for more.
+
+### Subscribing to events
+
+Event subscriptions themselves are relatively unchanged (although the data you can access/get back has changed a little). Before:
+
+```rust
+let api = ClientBuilder::new()
+    .build()
+    .await?
+    .to_runtime_api::<polkadot::RuntimeApi<DefaultConfig, PolkadotExtrinsicParams<DefaultConfig>>>();
+
+let mut event_sub = api.events().subscribe().await?;
+
+while let Some(events) = event_sub.next().await {
+    // ...
+}
+```
+
+Now, we simply swap the client out for our new one, and the rest is similar:
+
+```rust
+let api = OnlineClient::<PolkadotConfig>::new().await?;
+
+let mut event_sub = api.events().subscribe().await?;
+
+while let Some(events) = event_sub.next().await {
+    // ...
+}
+```
+
+See the `examples/examples/subscribe_all_events.rs` example for more.
+
+The general pattern, as seen above, is that we break apart constructing a query/address and using it. You can now construct queries dynamically instead and forego all static codegen by using the functionality exposed in the `subxt::dynamic` module instead.
+
+Other smaller breaking changes have happened, but they should be easier to address by following compile errors.
+
+For more details about all of the changes, the full commit history since the last release is as follows:
+
+### Added
+
+- Expose the extrinsic hash from TxProgress ([#614](https://github.com/paritytech/subxt/pull/614))
+- Add support for `ws` in `subxt-cli` ([#579](https://github.com/paritytech/subxt/pull/579))
+- Expose the SCALE encoded call data of an extrinsic ([#573](https://github.com/paritytech/subxt/pull/573))
+- Validate absolute path for `substitute_type` ([#577](https://github.com/paritytech/subxt/pull/577))
+
+### Changed
+
+- Rework Subxt API to support offline and dynamic transactions ([#593](https://github.com/paritytech/subxt/pull/593))
+- Use scale-decode to help optimise event decoding ([#607](https://github.com/paritytech/subxt/pull/607))
+- Decode raw events using scale_value and return the decoded Values, too ([#576](https://github.com/paritytech/subxt/pull/576))
+- dual license ([#590](https://github.com/paritytech/subxt/pull/590))
+- Don't hash constant values; only their types ([#587](https://github.com/paritytech/subxt/pull/587))
+- metadata: Exclude `field::type_name` from metadata validation ([#595](https://github.com/paritytech/subxt/pull/595))
+- Bump Swatinem/rust-cache from 1.4.0 to 2.0.0 ([#597](https://github.com/paritytech/subxt/pull/597))
+- Update jsonrpsee requirement from 0.14.0 to 0.15.1 ([#603](https://github.com/paritytech/subxt/pull/603))
```

Comment:
not sure how useful this is for subxt but with this jsonrpsee version

users can trace individual calls based on the tracing span such as if one only want to trace calls to "say_hello" by `RUST_LOG=jsonrpsee[method_call{name = "say_hello"}]=trace`